### PR TITLE
fix: killing the process hangs issue

### PR DESCRIPTION
**Pull request by helio-frota**
Related to #120

### CODE REVIEW

**Comment by helio-frota** [comment]
Steps to reproduce:

1) create a function
2) change the dependency to use this: 

```
"faas-js-runtime": "github:helio-frota/faas-js-runtime#replace-death"
```

3) Run the function
4) Try to access the function via browser (I'm using our electron example .. npm install , npm start)
5) kill the function with control + c

**Comment by helio-frota** [comment]
Video:


https://user-images.githubusercontent.com/6443576/216768709-e364141e-b262-4145-9c27-1168a81c283e.mp4

**Comment by helio-frota** [comment]
After merge the changes from main branch I started to see this in CI



I'm using Node 18 maybe is this something related to package-lock.json ?

**Comment by helio-frota** [comment]
> I'm using Node 18 maybe is this something related to package-lock.json ?

Changed to node 16x , deleted and re-generated `package-lock.json`, updated the PR and now the CI is working

**Comment by lance** [comment]
> @lance was there a reason for choosing the `death` module?

Nothing other than, "it seems to work as expected" 🤣

File: bin/cli.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ async function runServer(file) {
       console.error(code);
       throw TypeError(`Cannot find Invokable function 'handle' in ${code}`);
     }
-    ON_DEATH(_ => {
+    nodeCleanup(_ => {
       server.close();
```
Comment (lholmquist) [inline bin/cli.js:44]: I think the only thing that needs to be added is a `process.exit(0)` after the `server.close` instead of changing the module.

Comment (lance) [reply bin/cli.js:44]: @helio-frota can you give this a try in your test scenario?

Comment (helio-frota) [reply bin/cli.js:44]: @lholmquist @lance that works thanks!

**Comment by helio-frota** [comment]
PR sent https://github.com/nodeshift/faas-js-runtime/pull/172

**Pull request closed without merging**